docs/en/wallet-attestation.rst

```diff
@@ -30,271 +33,271 @@ The following requirements for the Wallet Attestation are met:
+
+   1. It MUST ensure that Wallet Hardware Keys do not already exist, if they exist and the Wallet is in the initialization phase they must be deleted.
+   2. It MUST generate a pair of asymmetric EC keys (Wallet Hardware Keys) via a local WSCD.
+   3. It SHOULD obtain a unique identifier (Wallet Hardware Key Tag) of the generated Wallet Hardware Keys from the operating system. If the operating system allows you to define a tag when creating keys, then a random string for Wallet Hardware Key Tag must be chosen.
+   4. It MUST store the relative Wallet Hardware Key Tag in a local storage.
+
+.. note::
+
+    **WSCD:** For key generation, the Wallet Instance will always rely on a local WSCD on devices that support it. For Android, Strongbox will be used or, if it is not available, a Trusted Execution Environment (TEE) while for iOS Secure Elements (SE). Since each OEM develops a different SDK for accessing the local WSCD, it will be treated in a generic manner below.
+
+
+**Step 7**: The Wallet Instance: invokes the operating system attestation service (Device Integrity Service) passing ``challenge`` and Wallet Hardware Key Tag in order to obtain a Key Attestation.
+
+.. note::
+
+    **Device Integrity Service:** Below we will treat a Device Integrity Service as an abstract entity made available by device OEMs which allows us to certify via a signed payload (attestation) that a particular key is hardware backed in a safe and protected environment of the device. Furthermore, the service will also allow you to demonstrate via a signed payload (assertion) that that particular app instance (Wallet Instance) is intact and genuine and has not been tampered with. Furthermore, both within the attestation and the assertion, the service includes essential information to recover the type of device, the model, the app version, the operating system version, the bootloader status and other information that allows you to understand whether the device has been tampered with or not. For Android the service used is `Key Attestation`_ in addition to `Play Integrity API`_, while for iOS the `DeviceCheck`_ service.
+
+**Step 8**: The Device Integrity Service:
+
+  * generate a Key Attestation holder binded with ``challenge`` and Wallet Hardware public key.
+  * adds relevant information on the security of the device.
+  * sign the Key Attestation with an OEM certificate which in addition to certifying the integrity of the Wallet Instance also certifies that the Wallet Hardware Keys is secure stored by the OS (hardware backed).
+
+**Step 9**: The Wallet Instance send the ``challenge`` with Key Attestation and Wallet Hardware Key Tag to the Wallet Provider Backend in order to register the Wallet Instance identified by the Wallet Hardware Key public key.
+
+.. note::
+
+  The Key Attestation (``key_attestation``) must always be encoded in base64 as the format is different between the various operating systems.
```

Comment:
```suggestion
  The Key Attestation (``key_attestation``) MUST be encoded in base64.
```

Reply:
if this is a requirement it should be taken out from a note and included in the general requirements